packages/verified-fetch/src/utils/parse-url-string.ts

```diff
@@ -207,7 +225,19 @@ export async function parseUrlString ({ urlString, ipns, logger }: ParseUrlStrin
         log.trace('Attempting to resolve DNSLink for %s', decodedDnsLinkLabel)
 
         try {
-          resolveResult = await ipns.resolveDNSLink(decodedDnsLinkLabel, options)
+          // eslint-disable-next-line max-depth
+          if (withServerTiming) {
+            const resolveResultWithServerTiming = await serverTiming('ipns.resolveDNSLink', `Resolve DNSLink ${decodedDnsLinkLabel}`, ipns.resolveDNSLink.bind(ipns, decodedDnsLinkLabel, options))
+            serverTimings.push(resolveResultWithServerTiming)
+            // eslint-disable-next-line max-depth
+            if (resolveResultWithServerTiming.error != null) {
+              throw resolveResultWithServerTiming.error
+            }
+            resolveResult = resolveResultWithServerTiming.result ?? undefined
```

Comment:
```suggestion
            resolveResult = resolveResultWithServerTiming.result
```

Reply:
this was needed for typescript and then I updated some code.. just checked if we can remove the null coalescing to undefined and it seems like we can.. just pushed up the change